src/components/Page/context.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import { useImmer } from "use-immer";
+import { filter, pipe, Source, Subject, subscribe } from "wonka";
+
+export interface Signal<T> {
+  type: "SIGNAL";
+
+  path: string;
+  interface: string;
+  member: string;
+  body: T;
+}
+
+export type Message = Signal<unknown>;
+
+interface Value {
+  bus: Subject<Message>;
+}
+
+const Context = React.createContext<undefined | Value>(undefined);
+export const Provider = Context.Provider;
+
+export function useContext(): Value {
+  const value = React.useContext(Context);
+  if (!value) {
+    throw new Error("…");
+  }
+
+  return value!;
+}
+
+export function useMailbox(id: string): Source<Message> {
+  const { bus } = useContext();
+
+  return React.useMemo(
+    () =>
+      pipe(
+        bus.source,
+        filter((x) => x.path === `/dev/timvir/component/${id}`)
+      ),
+    [bus, id]
+  );
+}
+
+export function send<T>(context: Value, id: string, member: string, body: T) {
+  context.bus.next({
+    type: "SIGNAL",
+    path: `/dev/timvir/component/${id}`,
```

Comment:
I struggle with hardcoding this like so, but I know you have a different opinion 😄

Reply:
That string is part of the API, as is the interface name (see https://timvir-git-mbox.timvir.vercel.app/tech/message-bus/interfaces).